js/modules/k6/metrics/metrics.go

```diff
@@ -109,12 +109,11 @@ func (m Metric) add(v goja.Value, addTags goja.Value) (bool, error) {
 	}
 
 	tags := state.Tags.GetCurrentValues()
-	if addTags != nil {
-		tagsobj := addTags.ToObject(m.vu.Runtime())
-		for _, key := range tagsobj.Keys() {
-			tags = tags.With(key, tagsobj.Get(key).String())
-		}
+	newTags, err := common.ApplyCustomUserTags(m.vu.Runtime(), tags, addTags)
+	if err != nil {
+		return false, fmt.Errorf("failed to apply the defined tags: %w", err)
```

Comment:
this error wrapping is more confusing to the user than helpful... they don't care that we internally "apply" tags, they are just passing a JS object with `{key: value}` tags. So, there either shouldn't be any error wrapping here, or it should be something specific for where the error originates, like this:
```suggestion
		return false, fmt.Errorf("invalid custom metric tags: %w", err)
```
or maybe even add the `Metric`'s name in the error? :thinking:

Reply:
```suggestion
		return false, fmt.Errorf("'%s' custom metric defines invalid tags: %w", m.metric.Name, err)
```
Changed with this